runtime/include/MemRefUtils.hpp

```diff
@@ -0,0 +1,143 @@
+// Copyright 2022-2023 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <cstring>
+
+#include <Exception.hpp>
+
+extern "C" {
+void *_mlir_memref_to_llvm_alloc(size_t size);
+void *_mlir_memref_to_llvm_aligned_alloc(size_t alignment, size_t size);
+void _mlir_memref_to_llvm_free(void *ptr);
+}
+
+template <typename T, size_t R> struct MemRefT {
+    T *data_allocated;
+    T *data_aligned;
+    size_t offset;
+    size_t sizes[R];
+    size_t strides[R];
+};
```

Comment:
This is still exposed in the public API, which we should avoid.

Reply:
What's your proposal?

Reply:
Fixed [d57fc34f7](https://github.com/PennyLaneAI/catalyst/pull/109/commits/170ab09406637677c830a808ab5fe21d57fc34f7)